### PR TITLE
A print for transition objects

### PR DESCRIPTION
**Pull request by dallenmidd**

### CODE REVIEW

**Comment by mylesstokowski** [comment]
@dallenmidd I know you're still working on this. I had some trouble getting the `life_cycle` to use the custom print method I had written when just calling some life cycle object, e.g. `config_ex_1$cycle` rather than `print.life_cycle(config_ex_1$cycle)`. I believe you may need to run `devtools::document()` which should add a line like `S3method(print,transition)` to the `NAMESPACE` file - I think that did it for me. The first answer in this [thread](https://stackoverflow.com/questions/48855993/defining-custom-print-methods-for-arrays-and-atomic-vectors) helped me get there.

**Comment by dallenmidd** [comment]
> @dallenmidd I know you're still working on this. I had some trouble getting the `life_cycle` to use the custom print method I had written when just calling some life cycle object, e.g. `config_ex_1$cycle` rather than `print.life_cycle(config_ex_1$cycle)`. I believe you may need to run `devtools::document()` which should add a line like `S3method(print,transition)` to the `NAMESPACE` file - I think that did it for me. The first answer in this [thread](https://stackoverflow.com/questions/48855993/defining-custom-print-methods-for-arrays-and-atomic-vectors) helped me get there.

Thanks for this! I will see whether I can figure this out when I am done with the `print`!

**Comment by dallenmidd** [comment]
I think that somehow I might have broken something? I tried your `devtools::document()` command and then re-building the package but now it won't the `run` command. It says that it can't find the `mutate` command. I assume this must be something about how we removed some of those `tidyverse` functions but I am still have the old version of `run`?? I am ready to merge this PR after I re-knit the readme. But I don't want to mess anythign up.

**Comment by mylesstokowski** [comment]
@dallenmidd just getting a chance to look over this. Basically, I think what happened is that in the time since you branched off `master`, there were a bunch of changes merged to `master`, like updating `run() ` and removing `tidyverse` libraries as you mention.

We'll need to address the merge conflicts before merging this, and make sure the tests are passing with `devtools::test()`. The basic workflow would look something like:
- Revert the changes to the `.Rd` files. These were updated with `devtools::document()`, which I'm realizing makes for more merge conflicts
- Merge the `master` branch into this branch and manually resolve any conflicts in the `.R` files
- Re-run `devtools::document()` and commit the updated, auto-generated documentation and `NAMESPACE`
- Ensure that `devtools::test()` is passing, and update any tests where the behavior has changed intentionally. Specifically, we might need to update any tests that include the printed representation of a `transition` 

Want me to take this? Happy to go either way.

**Comment by mylesstokowski** [comment]
@dallenmidd sounds good - will ping you back when I've made progress.

**Review by mylesstokowski — APPROVED** [review]
@dallenmidd I made the changes I described above. Now the diff between this branch and `master` looks right to me. And the tests all passed.

Much nicer print now, looks great to me! Feel free to merge this.

File: R/transition.R
Comment (mylesstokowski) [inline R/transition.R:194]: actually, just one idea: perhaps we could add this remove repeated whitespace in the function?
```suggestion
  function_string <- gsub(" +", " ", function_string)
```

**Pull request merged**
